### DISCUSSION TITLE
How do I correctly type a function like `empty` on a container class with generics?

### DISCUSSION THREAD

**Discussion by abadrinath15** [microsoft/pyright#5442] [Q&A] [answered]
Please consider the following assumptions:
```python
from dataclasses import dataclass
from typing import Generic, TypeVar

T = TypeVar("T")


@dataclass
class Queue(Generic[T]):
    forward: list[T]
    reverse: list[T]

```

Now, let's say we want to define a function (_not_ a method) `empty` to call on the `Queue`. 
Approximately I'd want the signature to be something like this:
```python
def empty(q) -> bool:
    ...
```

How should I type `q`? 
- `q: Queue`: pyright complains: `Expected type arguments for generic class "Queue"`
- `q: Queue[T]`:  pyright complains: `TypeVar "T" appears only once in generic function signature`

**Comment by erictraut** [comment, marked as answer]
If your function accepts a `Queue` with any type argument, then the correct type annotation for the `q` parameter is `Queue[Any]`.

**Reply by abadrinath15** [reply]
Duh. Can't believe I missed that. Thank you!
